bin/trace.ml

```diff
@@ -44,6 +44,28 @@ let base_of_sexp (sexp : Sexp.t) =
   | _ -> invalid_sexp sexp
 ;;
 
+let iter_sexps_follow file ~f =
+  Io.String_path.with_file_in ~binary:true file ~f:(fun chan ->
+    let rec loop () =
+      match Csexp.input_opt chan with
+      | Ok (Some sexp) ->
+        (* Check if exit event before processing *)
+        let is_exit =
+          match base_of_sexp sexp with
+          | "config", "exit", _, _ -> true
+          | _ -> false
+          | exception _ -> false
+        in
+        f sexp;
+        if not is_exit then loop ()
+      | Ok None | Error _ ->
```

Comment:
How do we exit `dune trace` in this mode before EOF?

Reply:
Wdym?